terraform/modules/tf_stream_alert_flow_logs/iam.tf

```diff
@@ -1,98 +1,126 @@
 // Allow flow logs to write to CloudWatch
 // http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/flow-logs.html#flow-logs-iam
-
+// IAM Role: Clustered VPC Flow Log
 resource "aws_iam_role" "flow_log_role" {
-  name = "flow_log_role"
-
-  assume_role_policy = <<EOF
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Sid": "",
-      "Effect": "Allow",
-      "Principal": {
-        "Service": "vpc-flow-logs.amazonaws.com"
-      },
-      "Action": "sts:AssumeRole"
+  name = "stream_alert_${var.cluster}_flow_log_role"
+
+  assume_role_policy = "${data.aws_iam_policy_document.flow_log_assume_role_policy.json}"
+}
+
+// IAM Policy Doc: AssumeRole for VPC Flow Logs
+data "aws_iam_policy_document" "flow_log_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["vpc-flow-logs.amazonaws.com"]
     }
-  ]
-} 
-EOF
+  }
 }
 
+// IAM Policy: CloudWatch Put Events
 resource "aws_iam_role_policy" "flow_log_write" {
-  name = "write_to_cloudwatch"
+  name = "CloudWatchPutEvents"
   role = "${aws_iam_role.flow_log_role.id}"
 
-  policy = <<EOF
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Action": [
-        "logs:CreateLogGroup",
-        "logs:CreateLogStream",
-        "logs:PutLogEvents",
-        "logs:DescribeLogGroups",
-        "logs:DescribeLogStreams"
-      ],
-      "Effect": "Allow",
-      "Resource": "*"
-    }
-  ]
-}   
-EOF
+  policy = "${data.aws_iam_policy_document.flow_log_put_cloudwatch_logs.json}"
+}
+
+// IAM Policy Doc: CloudWatch Put Events
+data "aws_iam_policy_document" "flow_log_put_cloudwatch_logs" {
+  statement {
+    actions = [
+      "logs:CreateLogGroup",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:DescribeLogGroups",
+      "logs:DescribeLogStreams",
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
```

Comment:
Nit: Add `effect: Allow`. It is the default, but let's put it to be explicit and consistent